File: fc-chat/demo/part3/ch4/message-system-add-push/src/main/java/net/prostars/messagesystem/dto/domain/Channel.java
package net.prostars.messagesystem.dto.domain;

import java.util.Objects;

public record Channel(ChannelId channelId, String title, int headCount) {

  @Override
  public boolean equals(Object o) {
    if (o == null || getClass() != o.getClass()) return false;
    Channel channel = (Channel) o;
    return Objects.equals(channelId, channel.channelId);
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(channelId);
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/domain/Channel.kt
package com.yscorp.withpush.messagesystem.dto.domain

import java.util.*

@JvmRecord
data class Channel(val channelId: ChannelId, val title: String, val headCount: Int) {
    override fun equals(o: Any?): Boolean {
        if (o == null || javaClass != o.javaClass) return false
        val channel = o as Channel
        return channelId == channel.channelId
    }

    override fun hashCode(): Int {
        return Objects.hashCode(channelId)
    }
}
